salt/state.py

```diff
@@ -1847,6 +1847,8 @@ def call(self, low, chunks=None, running=None, retries=1):
             if u'check_cmd' in low and u'{0[state]}.mod_run_check_cmd'.format(low) not in self.states:
                 ret.update(self._run_check_cmd(low))
             self.verify_ret(ret)
+            if isinstance(ret['comment'], list):
+                ret['comment'] = u'\n'.join(ret['comment'])
```

Comment:
I am going to approve this PR so long as the list of strings never leaves the state system. just to make things consistent with a ton of external software that relies on these data structures being consistent.
We should consider placing a comment to this effect in here.